t/op/require_errors.t

```diff
@@ -271,3 +271,15 @@ eval "require \"$nonsearch\"";
 
 like $@, qr/^Can't locate \Q$nonsearch\E at/,
         "correct error message for require $nonsearch";
+
+{
+    # make sure require doesn't treat a non-PL_sv_undef undef as
+    # success in %INC
+    # GH #17428
+    push @INC, "lib";
+    ok(!eval { require CannotParse; }, "should fail to load");
+    local %INC = %INC; # copies \&PL_sv_undef into a new undef
+    ok(!eval { require CannotParse; },
+       "check the second attempt also fails");
+    like $@, qr/Compilation failed in require/, "check we failed for the right reason";
```

Comment:
Should line 284 be `qr/Attempt to reload/` rather than `qr/Compilation failed in require/`, as the latter appears in both the first and second attempt at requiring, but the former only occurs on the second attempt.

Reply:
That's reasonable, I'll update it